### DISCUSSION TITLE
自动生成的tquic.h，导入的头文件不完整

### DISCUSSION THREAD

**Discussion by wsllf** [Tencent/tquic#242] [Q&A (问题和回答)]
![image](https://github.com/Tencent/tquic/assets/22959177/7f9e0208-f411-4c07-b357-d734d7681bf2)
这个SSL_CTX该引入哪个头文件

**Comment by iyangsj** [comment]
需包含 BoringSSL库的头文件
```
#include "openssl/ssl.h"
```

参考示例程序：
https://github.com/tquic-group/tquic-example-c/blob/main/simple_client.c#L29

**Reply by wsllf** [reply]
在集成时，需要把BoringSSL单独编译一份集成到项目里吗，还是说 tquic 项目已经在编译结果 libtquic.so 中包含了 SSL 的实现

**Reply by iyangsj** [reply]
编译时要链接BoringSSL库

**Comment by wsllf** [comment]
你好，如下图，我在连接连接成功的时候，创建了双向流，然后在流可写的回调里，进行了写入流发送操作，服务端也接收到了我发送的数据，服务端给我回复了OK，但是我这客户端没有回调client_on_stream_readable方法，请问这可能是什么问题，服务端用的是tquic-example-c项目的server demo
![image](https://github.com/Tencent/tquic/assets/22959177/2c115c1c-2ae9-4ad0-9766-161e632bd940)
![image](https://github.com/Tencent/tquic/assets/22959177/eefd51ac-519b-4511-87ac-d7a3a9524b5a)
![image](https://github.com/Tencent/tquic/assets/22959177/787ce90f-0439-4b06-a0d5-40b8e9e427d5)
![image](https://github.com/Tencent/tquic/assets/22959177/158f7a42-a209-4eed-ab3f-5cf910d6cb58)

**Reply by iyangsj** [reply]
@wsllf 把服务端debug或trace级别的日志 和 客户端相关代码 通过附件发出来看看

**Reply by wsllf** [reply]
客户端发送stream时， quic_stream_write方法的fin参数设置为true之后，可以接收到服务端发送的数据了，请问下这个fin参数是什么含义

**Reply by wsllf** [reply]
刚才查了下quic资料，描述了这个fin参数和tcp的类似，代表流是否发送完成

**Reply by iyangsj** [reply]
fin为true代表结束流的写入。客户端代码正确处理quic_stream_wantread()了吗？
可以把前后两次client的debug级别日志发来看看

**Reply by wsllf** [reply]
问题已经解决了，服务端是判断客户端流发送finish才给客户端回复的数据，客户端之前没注意这个参数，导致服务端一直没回复消息
